benchmarking-tests/benchmark_utils.py

```diff
@@ -36,6 +44,38 @@ def run_set_statsd_host(args):
         json.dump(conf, f)
 
 
+def run_gather_metrics(args):
+    # setup a rethinkdb connection
+    conn = r.connect(args.bigchaindb_host, 28015, 'bigchain')
+
+    # setup csv writer
+    csv_file = open(args.csvfile, 'w')
+    csv_writer = csv.writer(csv_file)
+
+    # query for the number of transactions on the backlog
+    num_transactions = r.table('backlog').count().run(conn)
+    logger.info('Starting gathering metrics. {} transasctions in the backlog'.format(num_transactions))
+
+    # listen to the changefeed
+    try:
+        for change in r.table('bigchain').changes().run(conn):
+            # check only for new blocks
+            if change['old_val'] is None:
+                block_num_transactions = len(change['new_val']['block']['transactions'])
+                time_now = time.time()
+                logger.info('{} {}'.format(time_now, block_num_transactions))
+                csv_writer.writerow([str(time_now), str(block_num_transactions)])
+
+                num_transactions -= block_num_transactions
+                if num_transactions == 0:
```

Comment:
There's an implicit assumption here that all the transactions are valid, so each one will never-again be seen in the backlog for inclusion in a future block. A more general check would be to query the backlog to check how many transactions are in it (i.e. `r.table('backlog').count().run(conn)`)... but that adds some querying overhead.